src/models/transfer.py

```diff
@@ -85,6 +87,11 @@ def summarize(transfers: list[Transfer]) -> str:
             f"Total COW Funds needed: {cow_total / 10 ** 18:.4f}\n"
         )
 
+    @property
+    def recipient(self) -> Address:
+        """The correct way to access the true recipient of a transfer"""
+        return self.redirect_to if self.redirect_to is not None else self.receiver
```

Comment:
I think a lot of your concerns here can be resolved by defining a proper constructor here that only allows users to specify recipient and making the two fields receiver and redirect to private. I did actually plan to do this but forgot to come back to it. 

If we're go this route, we would then be making "try-redirect" the only way of actually redirecting transfers and all of it tests would become cumbersome to setup, but I do believe this would be the most correct and safe approach. Furthermore, making these fields private would help us catch places where they might be used outside of the class and encourage users to always use "recipient".

Reply:
Fixed.